hibernate-core/src/main/java/org/hibernate/tool/schema/spi/SchemaManagementTool.java

```diff
@@ -32,4 +43,25 @@
 	 * @param generationTarget the custom instance to use.
 	 */
 	void setCustomDatabaseGenerationTarget(GenerationTarget generationTarget);
+
+	default ExtractionContext createExtractionContext(
+			ServiceRegistry serviceRegistry,
+			JdbcEnvironment jdbcEnvironment,
+			DdlTransactionIsolator ddlTransactionIsolator,
+			Identifier defaultCatalog,
+			Identifier defaultSchema,
+			ExtractionContext.DatabaseObjectAccess databaseObjectAccess) {
+		return new ImprovedExtractionContextImpl(
+				serviceRegistry,
+				jdbcEnvironment,
+				ddlTransactionIsolator,
+				defaultCatalog,
+				defaultSchema,
+				databaseObjectAccess
+		);
+	}
+
+	default InformationExtractor createInformationExtractor(ExtractionContext extractionContext) {
+		return new InformationExtractorJdbcDatabaseMetaDataImpl( extractionContext );
+	}
```

Comment:
what about creating an interface `org.hibernate.tool.schema.extract.spi.ExtractionTool` and add this two methods there?

Reply:
@dreab8, I was thinking about something like that as well. I think that `#setCustomDatabaseGenerationTarget` could be included as well. I ended up creating a new `ReactiveSchemaManagementTool` in https://github.com/hibernate/hibernate-reactive/pull/903, so I think it would work well to encapsulate this stuff in an interface.

Before making that change, I'd like to take a look at `SchemaValidator` to see what is needed there in case something else needs to be added.

Reply:
@dreab8, thinking more about it, `ExtractionTool#getDatabaseGenerationTarget` doesn't make sense. I'm leaning more towards `SchemaManagementToolContext`. I'm doing some more refactoring and will have an update soon.

Reply:
@dreab8, there are complications with my idea, so I have added a [commit](https://github.com/hibernate/hibernate-orm/pull/4106/commits/e4390a014c8f189e57c20261b8bc30e8edf50958) that moves those methods into `ExtractionTool` as you suggested.

Reply:
@dreab8, I didn't create default methods in `ExtractionTool` because all of the interfaces are annotated with `@Incubating`.  Is that OK with you?

Reply:
Hi @gbadner , it looks good to me :+1: